src/data/data-store.ts

```diff
@@ -0,0 +1,61 @@
+import { Reference } from "../model/reference.model";
+
+/**
+ * This class is responsible for storing citation data and providing it when requested
+ */
+export class DataStore {
+    
+    /**
+     * Array containing reference objects
+     */
+    private static references: Reference[] = [];
+
+    /**
+     * Private constructor to prevent the instantiation of this class
+     */
+    private constructor () { }
+
+    /**
+     * Sets the data to be stored in the references array
+     * Runs in O(n), where n = this.references.length
+     * @param refs array of reference objects
+     */
+    public static setReferences (refs: Reference[]): void {
+        this.references = [];
+
+        /*
+          Why not just do 'this.references = refs'?
+          In this case, refs and this.references will point to the same object in memory,
+          so any changes applied to refs outside the class will be reflected in this.references and vice versa
+          This violates encapsulation and might cause unexpected behavior.
+          The safer option is to push every element from refs to this.references
+         */
+        refs.forEach(ref => this.references.push(ref));
```

Comment:
Hmm, I think you're indeed over-engineering here, and even worse you're not over-engineering enough :-) Because if a reference object is modified, your data will also be modified. If you want to prevent that you'd have to deep clone each reference object.

If you really want to deal with immutable data, perhaps you should use a library from the start such as immutable.js or immer (used by Joplin). That's a reasonable approach and probably more correct since, as you can see, even though you're trying to go for immutable data, you're making a mistake and the data is not immutable. The lib would prevent such mistakes by freezing the data.

It's up to you I think. It can be interesting to learn about this, but perhaps not necessary for this project.